tests/optim/test_multi.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import absolute_import, division, print_function
+
+import pytest
+import torch
+from torch.distributions import constraints
+
+import pyro
+import pyro.distributions as dist
+import pyro.optim
+import pyro.poutine as poutine
+from pyro.optim.multi import MixedMultiOptimizer, Newton2d, PyroMultiOptimizer, TorchMultiOptimizer
+from tests.common import assert_equal
+
+FACTORIES = [
+    lambda: PyroMultiOptimizer(pyro.optim.Adam({'lr': 0.05})),
+    lambda: TorchMultiOptimizer(torch.optim.Adam, {'lr': 0.05}),
```

Comment:
Should we test more Pyro and Torch optimizers?  Are there any we'd expect to fail (LBFGS?)?